upup/pkg/fi/cloudup/hetznertasks/loadbalancer.go

```diff
@@ -242,20 +263,26 @@ func (_ *LoadBalancer) RenderHetzner(t *hetzner.HetznerAPITarget, a, e, changes
 						return err
 					}
 				}
+				_, errCh := actionClient.WatchProgress(ctx, action)
+				if err := <-errCh; err != nil {
+					return err
+				}
 			}
 		}
 
-	}
-
-	// Add the target separately, otherwise UsePrivateIP cannot be set
-	// https://github.com/hetznercloud/hcloud-go/pull/198
-	if a == nil || a.Target == "" {
-		_, _, err := client.AddLabelSelectorTarget(context.TODO(), loadbalancer, hcloud.LoadBalancerAddLabelSelectorTargetOpts{
-			Selector:     e.Target,
-			UsePrivateIP: fi.Bool(true),
-		})
-		if err != nil {
+		// Update the targets
+		if a.Target == "" {
+			action, _, err := client.AddLabelSelectorTarget(ctx, loadbalancer, hcloud.LoadBalancerAddLabelSelectorTargetOpts{
+				Selector:     e.Target,
+				UsePrivateIP: fi.Bool(true),
+			})
 			if err != nil {
+				if err != nil {
+					return err
+				}
+			}
+			_, errCh := actionClient.WatchProgress(ctx, action)
+			if err := <-errCh; err != nil {
```

Comment:
Nit: we might want a helper here that calls WatchProgress, but also has a timer, so that we log e.g. every minute that we're waiting for something, rather than potentially silently waiting forever.

Reply:
Good idea. Thanks!